frontend-web/webclient/app/Applications/Jobs/Widgets/GenericFiles.tsx

```diff
@@ -52,17 +51,26 @@ export const FilesParameter: React.FunctionComponent<FilesProps> = props => {
 
     const onActivate = useCallback(() => {
         const pathRef = {current: ""};
+        const provider = getProviderField();
         dialogStore.addDialog(
             <FilesBrowse
                 browseType={BrowseType.Embedded}
                 pathRef={pathRef}
+                additionalFilters={provider ? {
+                    filterProvider: provider
+                } : undefined}
```

Comment:
I don't remember what we agreed on. Should this behavior remain, or should it instead disallow selection with explanation in FileSelector?